src/PageContent/RequestPickup/RecurringPickupRequest.js

```diff
@@ -178,113 +181,221 @@ class RecurringPickupRequest extends Component {
         if (!this.handleValidation()) {
             alert('Form has errors');
         } else {
-            // force request's timezone to be the same as DA's
-            let reqTimezone = this.props.donatingAgency.timezone;
-            let dateTimeStringToTimestamp = (dateString, timeString) =>
-                moment
-                    .tz(
-                        dateString + timeString,
-                        InputFormat.DATE + InputFormat.TIME,
-                        reqTimezone
-                    )
-                    .valueOf();
-            let startTimestamp = dateTimeStringToTimestamp(
-                event.target.startDate.value,
-                event.target.startTime.value
-            );
+            var deliveryRequest;
+            var raInfo = {};
+            var raRequested = null;
+            var raIndex = event.target.receivingAgency.value;
+            var dgInfo = {};
+            var dgRequested = null;
+            var dgIndex = event.target.delivererGroup.value;
+            var primaryContact = this.state.memberList[
+                event.target.primaryContact.value
+            ];
 
-            let durationValue;
-            // compute ending Timestamp
-            let endTimestamp;
-            if (
-                event.target.endCriteria.value === RequestEndCriteriaType.DATE
-            ) {
-                durationValue = event.target.endDate.value;
-                endTimestamp = dateTimeStringToTimestamp(
-                    durationValue,
+            if (this.props.type === DeliveryType.RECURRING) {
```

Comment:
Are the changes you made in this function basically:
```
if (this.props.type === DeliveryType.RECURRING) {
    all the original logic
} else {
    a copy of the original logic + some other fields like food items
}
```
?
If so, please make sure you didn't change anything for recurring pickup requests, and just leave the emergency pickup request part empty for backend dev to fill it in (the schema is different and has just been decided today).

Reply:
Yep, thats exactly how I did it